packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -852,6 +936,10 @@ class _SelectableRegionState extends State<SelectableRegion> with TextSelectionD
   @override
   void bringIntoView(TextPosition position) {/* SelectableRegion must be in view at this point. */}
 
+  @Deprecated(
```

Comment:
Looks like we're getting there actually...  I've created an issue to track this (https://github.com/flutter/flutter/issues/111213) and I'm going to link it in your TODO.

Reply:
Is it a lot of work to migrate? Since this class is previously private and this pr makes it public, should we do it as part of this pr?

Reply:
I was worried it was going to be, but I tried it and I think it worked out super nicely: https://github.com/flutter/flutter/pull/107193/commits/4cee48b3d86508e0c7cc0c2c74061d4cd13d0eee

So the TextSelectionDelegate parameter of SelectionOverlay is now deprecated, which is great, and TextSelectionDelegate is removed from SelectableRegion.  Thanks for calling this out.

Reply:
Actually I spoke too soon.  In order to keep the old way of doing things working, I need to keep TextSelectionDelegate there during the deprecation period.  I reverted that commit but kept the deprecations.  After we remove the deprecations, it will be really easy to remove TextSelectionDelegate for good.

Reply:
gotcha